lib/streams.gi

```diff
@@ -1388,6 +1388,29 @@ InstallGlobalFunction( InputFromUser,
   end );
 
 
+#############################################################################
+##
+#M  OpenExternal(filename)  . . . . . . . . . . . . open file in external GUI
+##
+InstallGlobalFunction( OpenExternal, function(filename)
+    local file;
+    if ARCH_IS_MAC_OS_X() then
+      Exec(Concatenation("open \"",filename,"\""));
```

Comment:
OK, I don't want to whine, but I have to repeat one of my pet mantras, namely that "Exec is a really bad function, we should never have added it like that". Besides not even allowing to check the return value, it also forces us to do silly quoting tricks like the above; and this is not even complete, a filename with a `"` in it will break it. We really should think about adding an alternative to `Exec` which is a bit better (doesn't go through a shell and hence has no quoting issues, and gives access to return value, and optionally allows for suppressing or redirecting output.

Reply:
(But of course none of this is required for this PR!)